src/opcode.rs

```diff
@@ -1610,6 +1610,43 @@ opcode!(
     }
 );
 
+opcode!(
+    /// Send a zerocopy message on a socket, equivalent to `send(2)` using a pre-mapped buffer that
+    /// has been previously registered with
+    /// [`Submitter::register_buffers`](crate::Submitter::register_buffers).
+    pub struct SendZcFixed {
+        /// The `buf_index` is an index into an array of fixed buffers,
+        /// and is only valid if fixed buffers were registered.
+        ///
+        /// The buf and len arguments must fall within a region specified by buf_index in the
+        /// previously registered buffer. The buffer need not be aligned with the start of the
+        /// registered buffer.
+        fd: { impl sealed::UseFixed },
+        buf: { *const u8 },
+        len: { u32 },
+        buf_index: { u16 },
+        ;;
+        flags: i32 = 0,
+        zc_flags: u16 = 0,
+    }
+
+    pub const CODE = sys::IORING_OP_SEND_ZC;
+
+    pub fn build(self) -> Entry {
+        let SendZcFixed { fd, buf, len, buf_index, flags, zc_flags } = self;
+
+        let mut sqe = sqe_zeroed();
+        sqe.opcode = Self::CODE;
+        assign_fd!(sqe.fd = fd);
+        sqe.__bindgen_anon_2.addr = buf as _;
+        sqe.len = len;
+        sqe.__bindgen_anon_3.msg_flags = flags as _;
+        sqe.__bindgen_anon_4.buf_index = buf_index;
+        sqe.ioprio = zc_flags | sys::IORING_RECVSEND_FIXED_BUF as u16;
```

Comment:
It looks like kernel doesn't like using separate opcodes anymore, maybe we can add an optional method on `SendZc` to follow this style?

Reply:
Sure. `buf_index` or `fixed_buf_index`?

Reply:
But a value of 0 is valid buf_index. So not quite sure how to tell. Maybe set it to the max by default? Or allow negative -1
or wrap the type in `Other`?

Reply:
I think can use `Option<>`

Reply:
buf_index is fine, need some document.

Reply:
Done.